packages/manager/src/hooks/useAdobeAnalytics.ts

```
@@ -0,0 +1,53 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { ADOBE_ANALYTICS_URL, NUM_ADOBE_SCRIPTS } from 'src/constants';
+import { reportException } from 'src/exceptionReporting';
+
+import { loadScript } from './useScript';
+
+/**
+ * Initializes our Adobe Analytics script on mount and subscribes to page view events.
+ */
+export const useAdobeAnalytics = () => {
+  const history = useHistory();
+
+  React.useEffect(() => {
+    // Load Adobe Analytics Launch Script
+    if (!!ADOBE_ANALYTICS_URL) {
+      loadScript(ADOBE_ANALYTICS_URL, { location: 'head' })
+        .then((data) => {
+          const adobeScriptTags = document.querySelectorAll(
+            'script[src^="https://assets.adobedtm.com/"]'
+          );
+          // Log an error; if the promise resolved, the _satellite object and 3 Adobe scripts should be present in the DOM.
+          if (
+            data.status !== 'ready' ||
+            !(window as any)._satellite ||
+            adobeScriptTags.length !== NUM_ADOBE_SCRIPTS
+          ) {
+            reportException(
+              'Adobe Analytics error: Not all Adobe Launch scripts and extensions were loaded correctly; analytics cannot be sent.'
+            );
+          }
+        })
+        .catch(() => {
+          // Do nothing; a user may have analytics script requests blocked.
```

Am wondering if anyone would care to see events for that.

---

Not sure. I personally don't mind the silent failure

---

Silent for the user for sure. Just wondering about collecting this info on our end so we have a better idea of the sample for those events but it would be a product decision and I doubt anyone would be interested 🤷

---

Silent for the user for sure. Just wondering about collecting this info on our end so we have a better idea of the sample for those events but it would be a product decision and I doubt anyone would be interested 🤷

---

We could capture a sentry exception, but I worry we'd just flood sentry with useless messages. I have a feeling lots of browsers with privacy features enabled will block Adobe Analytics

---

^^ Yes, that was the problem before, though I think the error message was at a slightly different spot in the code at the time. We don't want to flood Sentry, and before we were creating a new Sentry alert for every user and every event they clicked where analytics were disabled. Where the `catch` is now, I think we'd just be triggering sentry alerts for every user blocking extensions every time they load the app, but that's still a lot.